app/src/androidTest/java/vn/mbm/phimp/me/MainActivityTest.java

```diff
@@ -0,0 +1,238 @@
+                                                0)),
+                                0),
+                        isDisplayed()));
+        textView4.check(matches(withText("Accounts")));
+
+        ViewInteraction uploadIconClick = onView(
+                allOf(withId(R.id.tab_upload), isDisplayed()));
+        uploadIconClick.perform(click());
+
+        Espresso.unregisterIdlingResources(idlingResource1);
+
+        IdlingPolicies.setMasterPolicyTimeout(waitingTime * 2, TimeUnit.MILLISECONDS);
+        IdlingPolicies.setIdlingResourceTimeout(waitingTime * 2, TimeUnit.MILLISECONDS);
+
+        IdlingResource idlingResource2 = new ElapsedTimeIdlingResource(waitingTime);
+        Espresso.registerIdlingResources(idlingResource2);
+
+        ViewInteraction textView2 = onView(
+                allOf(withId(R.id.textView1), withText("Accounts"),
+                        childAtPosition(
+                                allOf(withId(R.id.relativeLayout1),
+                                        childAtPosition(
+                                                withId(R.id.tabUpload),
+                                                0)),
+                                0),
+                        isDisplayed()));
+        textView2.check(matches(withText("Accounts")));
+
+        ViewInteraction imageButton = onView(
+                allOf(withId(R.id.btnUploadAccountAdd),
+                        childAtPosition(
+                                allOf(withId(R.id.relativeLayout1),
+                                        childAtPosition(
+                                                withId(R.id.tabUpload),
+                                                0)),
+                                1),
+                        isDisplayed()));
+        imageButton.check(matches(isDisplayed()));
+
+        ViewInteraction textView3 = onView(
+                allOf(withId(R.id.textView2), withText("Photos"),
+                        childAtPosition(
+                                allOf(withId(R.id.relativeLayout2),
+                                        childAtPosition(
+                                                withId(R.id.tabUpload),
+                                                3)),
+                                0),
+                        isDisplayed()));
+        textView3.check(matches(withText("Photos")));
+
+        ViewInteraction imageButton2 = onView(
+                allOf(withId(R.id.upload_sendDirectly),
+                        childAtPosition(
+                                allOf(withId(R.id.relativeLayout2),
+                                        childAtPosition(
+                                                withId(R.id.tabUpload),
+                                                3)),
+                                2),
+                        isDisplayed()));
+        imageButton2.check(matches(isDisplayed()));
+
+        Espresso.unregisterIdlingResources(idlingResource2);
+
```

Comment:
This is a very long test. It'd be awesome if this could be broken down into multiple short ones. Please do so if possible, and if not, let it be.